src/main/scala/ing/wbaa/druid/definitions/PostAggregation.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ing.wbaa.druid
+package definitions
+
+import ca.mrvisser.sealerate
+import io.circe._
+import io.circe.generic.auto._
+import io.circe.syntax._
+
+sealed trait PostAggregationType extends Enum with CamelCaseEnumStringEncoder
+object PostAggregationType extends EnumCodec[PostAggregationType] {
+  case object Arithmetic extends PostAggregationType
+
+  val values: Set[PostAggregationType] = sealerate.values[PostAggregationType]
+}
+
+trait PostAggregation {
+  val `type`: PostAggregationType
+  val name: String
+}
+
+object PostAggregation {
+  implicit val encoder: Encoder[PostAggregation] = new Encoder[PostAggregation] {
+    final def apply(pAgg: PostAggregation): Json =
+      (pAgg match {
+        case x: ArithmeticPostAggregation => x.asJsonObject
+      }).add("type", pAgg.`type`.asJson).asJson
+  }
+}
+
+case class ArithmeticPostAggregation(name: String, fn: String, fields: (Field, Field))
+    extends PostAggregation {
+  override val `type` = PostAggregationType.Arithmetic
+}
+
+sealed trait FieldType extends Enum with CamelCaseEnumStringEncoder
+object FieldType extends EnumCodec[FieldType] {
+  case object FieldAccess extends FieldType
+  case object Constant    extends FieldType
+
+  val values: Set[FieldType] = sealerate.values[FieldType]
+}
+
+trait Field {
```

Comment:
`Field` sounds pretty generic. If it is something that is only to be applied for `PostAggregation`s I would move the `FieldType` and `Field` objects within the `PostAggregation` companion object (to clarify this)

If this can be reused for other structures as well, I'd propose to move this to a separate file.

What do you think?

Reply:
Looking into the [documentation](http://druid.io/docs/latest/querying/post-aggregations.html#field-accessor-post-aggregators) more carefully, those structures should not be called `Field` but rather `PostAggregator`.

Since `PostAggregator` is more specific, would it be ok with you if we used that no name the structure instead?

Also, it looks like it is specific of the aggregation context. Would you recommend to keep it in the `Aggregations` file or to move it to its own file?